File: JavaCore_TestingSystem_Assignment_4/src/com/vti/entity/Account.java
package com.vti.entity;

import java.time.LocalDate;

public class Account {
	private int id;
	private String email;
	private String username;
	private String fullName;
	private Department department;
	private Position position;
	private LocalDate createDate;
	private Group[] groups;

	public Account() {
	}

	public Account(int id, String email, String username, String firstName, String lastName) {
		this.id = id;
		this.email = email;
		this.username = username;
		this.fullName = firstName + " " + lastName;
	}

	public Account(int id, String email, String username, String firstName, String lastName, Position position) {
		this(id, email, username, firstName, lastName);
		this.position = position;
		this.createDate = LocalDate.now();
	}

	public Account(int id, String email, String username, String firstName, String lastName, Position position,
			LocalDate createDate) {
		this(id, email, username, firstName, lastName, position);
		this.createDate = createDate;
	}

	// Constructor cho Question 3 (Group)
	public Account(String username) {
		this.username = username;
	}

	// Getter/Setter
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getFullName() {
		return fullName;
	}

	public void setFullName(String fullName) {
		this.fullName = fullName;
	}

	public Department getDepartment() {
		return department;
	}

	public void setDepartment(Department department) {
		this.department = department;
	}

	public Position getPosition() {
		return position;
	}

	public void setPosition(Position position) {
		this.position = position;
	}

	public LocalDate getCreateDate() {
		return createDate;
	}

	public void setCreateDate(LocalDate createDate) {
		this.createDate = createDate;
	}

	public Group[] getGroups() {
		return groups;
	}

	public void setGroups(Group[] groups) {
		this.groups = groups;
	}

	@Override
	public String toString() {
		return "Account{" + "id=" + id + ", email='" + email + "', username='" + username + "', fullName='" + fullName
				+ "', position=" + position + ", createDate=" + createDate + "}";
	}
}

File: Fontend_HTML_CSS/ReactJS/1. TypeScript/LAB2_Testing_System_OOP/Account.js
class Account {
    // Khai báo các thuộc tính.
    id;
    email;
    userName;
    fullName;
    department;
    position;
    createDate;
    // Khai báo constructor.
    constructor(id, email, userName, fullName, department, position, createDate) {
        this.id = id;
        this.email = email;
        this.userName = userName;
        this.fullName = fullName;
        this.department = department;
        this.position = position;
        this.createDate = createDate;
    }
    getId() {
        return this.id;
    }
    getEmail() {
        return this.email;
    }
    getUserName() {
        return this.userName;
    }
    getFullName() {
        return this.fullName;
    }
    getDepartment() {
        return this.department;
    }
    getPosition() {
        return this.position;
    }
    getCreateDate() {
        return this.createDate;
    }
    setId(id) {
        this.id = id;
    }
    setEmail(email) {
        this.email = email;
    }
    setUserName(userName) {
        this.userName = userName;
    }
    setFullName(fullName) {
        this.fullName = fullName;
    }
    setDepartment(department) {
        this.department = department;
    }
    setPosition(position) {
        this.position = position;
    }
    setCreateDate(createDate) {
        this.createDate = createDate;
    }
}
// Hàm in thông tin Account, viết bên ngoài class
function printInforAccount(account) {
    console.log("ID: " + account.getId() + " UserName: " + account.getUserName() + " Fullname: " + account.getFullName() + " Department: " + account.getDepartment().getName() + " Position: " + account.getPosition().getName());
}
// Export Account và hàm printInforAccount sang class khác để sử dụng.
export { Account, printInforAccount };
